lnwallet/script_utils.go

```diff
@@ -314,8 +313,10 @@ func SenderHtlcSpendRevoke(signer Signer, signDesc *SignDescriptor,
 
 	// Derive the revocation key using the local revocation base point and
 	// commitment point.
-	revokeKey := DeriveRevocationPubkey(signDesc.PubKey,
-		signDesc.DoubleTweak.PubKey())
+	revokeKey := DeriveRevocationPubkey(
+		signDesc.KeyDesc.PubKey,
```

Comment:
here we use the pubkey without checking it for nil-ness. Maybe adda check, or make `DeriveRevocationPubkey` return error in that case.

Reply:
Fixed.